server/models/User.js

```diff
@@ -0,0 +1,46 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const UserSchema = new Schema({
+    userName: {
+        type: String,
+        unique: true,
+        required: [true, 'Please provide a user name'],
+        maxLength: [12, 'User name cannot exceed 12 characters']
+    },
+    password:{
+        type: String,
+        required: [true, 'please provide a password'],
```

Comment:
Just curious, for password and email, should we add in validation?

Reply:
we can do some regex to make sure that there are the generally accepted requirements (at least one lower case, one upper case letter, one number, etc.) Not sure if we want to do that in the schema or if we want to nest that into the registration for the user

Reply:
Thanks for letting me know, Chris.